No `impl From<RelativePath> for RelativePathBuf`

It's my understanding that this library tries to match the functionality of the standard library's paths, yet, while the following works with the standard library's paths:

```rust
use std::path::{Path, PathBuf};

fn main() {
    let p = Path::new("foo/bar");
    let pb = PathBuf::from(p);
}
```

this doesn't work with `relative-path`:

```rust
use relative_path::{RelativePath, RelativePathBuf};

fn main() {
    let p = RelativePath::new("foo/bar");
    let pb = RelativePathBuf::from(p);
}
```

Error received:

```
error[E0277]: the trait bound `RelativePath: AsRef<str>` is not satisfied
 --> src/main.rs:5:14
  |
5 |     let pb = RelativePathBuf::from(p);
  |              ^^^^^^^^^^^^^^^^^^^^^ the trait `AsRef<str>` is not implemented for `RelativePath`
  |
  = help: the trait `AsRef<RelativePath>` is implemented for `RelativePath`
  = note: required because of the requirements on the impl of `From<&RelativePath>` for `RelativePathBuf`

For more information about this error, try `rustc --explain E0277`.
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -828,6 +828,12 @@ impl AsRef<RelativePath> for RelativePathBuf {
     }
 }
 
+impl AsRef<str> for RelativePath {
+    fn as_ref(&self) -> &str {
+        &self.inner
+    }
+}
+
 impl Borrow<RelativePath> for RelativePathBuf {
     fn borrow(&self) -> &RelativePath {
         self.deref()

--- a/src/tests.rs
+++ b/src/tests.rs
@@ -625,6 +625,11 @@ fn test_from() {
         RelativePathBuf::from(String::from("foo/bar")),
     );
 
+    assert_eq!(
+        RelativePathBuf::from(rp("foo/bar")),
+        RelativePathBuf::from("foo/bar"),
+    );
+
     assert_eq!(rp("foo/bar").to_owned(), RelativePathBuf::from("foo/bar"),);
 
     assert_eq!(&*Box::<RelativePath>::from(rp("foo/bar")), rp("foo/bar"));
@@ -644,6 +649,11 @@ fn test_from() {
         &*Rc::<RelativePath>::from(RelativePathBuf::from("foo/bar")),
         rp("foo/bar")
     );
+}
+
+#[test]
+fn test_relative_path_asref_str() {
+    assert_eq!(<RelativePath as AsRef<str>>::as_ref(rp("foo/bar")), "foo/bar");
 }
 
 #[test]

